docs/machine-learning/how-does-mldotnet-work.md

```diff
@@ -0,0 +1,262 @@
+            [ColumnName("Score")]
+            public float Price { get; set; }
+        }
+    
+        static void Main(string[] args)
+        {
+            MLContext mlContext = new MLContext();
+    
+            // 1. Import or create training data
+            HouseData[] houseData = {
+                new HouseData() { Size = 1.1F, Price = 1.2F },
+                new HouseData() { Size = 1.9F, Price = 2.3F },
+                new HouseData() { Size = 2.8F, Price = 3.0F },
+                new HouseData() { Size = 3.4F, Price = 3.7F } };
+            IDataView trainingData = mlContext.Data.LoadFromEnumerable(houseData);
+
+            // 2. Specify data preparation and model training pipeline
+            var pipeline = mlContext.Transforms.Concatenate("Features", new[] { "Size" })
+                .Append(mlContext.Regression.Trainers.Sdca(labelColumnName: "Price", maximumNumberOfIterations: 100);
+    
+            // 3. Train model
+            var model = pipeline.Fit(trainingData);
+    
+            // 4. Make a prediction
+            var size = new HouseData() { Size = 2.5F };
+            var predEngine = model.CreatePredictionEngine<HouseData, Prediction>(mlContext);
+            var price = predEngine.Predict(size);
+    
+            Console.WriteLine($"Predicted price for size: {size.Size*1000} sq ft= {price.Price*100:C}k");
+
+            // Predicted price for size: 2500 sq ft= $261.98k
+        }
+    } 
+```
+
+The following diagram shows this flow in visual form, with a couple of additions, which we talk about later in this article:
+
+- Evaluate the model and iterate to improve
+- Separate model development and model usage (deployment)
+    
+![ML.NET application development flow including components for data generation, pipeline development, model training, model evaluation, and model usage](./media/mldotnet-annotated-workflow.png) 
+
+Let's dig a little deeper into those concepts.
+
+## ML.NET machine learning model
+
+An ML.NET model is an encapsulation of the transformations to perform on your input data to arrive at the predicted output.
+
+### Simple
+
+The simplest model is two-dimensional linear regression, where one continuous quantity is proportional to another, as in the house price example above. 
+
+![Linear Regression Model with bias and weight parameters](./media/linear-regression-model.svg)
+
+The model is simply: `Price = b + Size * w`. The parameters `b` and `w` are estimated by fitting a line on a set of (size, price) pairs. The data used to find the parameters of the model is called **training data**. The inputs of a machine learning model are called **features**. In this example, `Size` is the only feature. The ground-truth values used to train a machine learning model are called **labels**. Here, the `Price` values in the training data set are the labels. 
+
+### More complex
+
+A more complex model classifies financial transactions into categories using the text description of the transaction.
```

Comment:
A more complex model classifies financial transactions into categories using the transaction text description.